drivers/samsung/displays/mdc_protocol.cr

```diff
@@ -0,0 +1,376 @@
+module Samsung; end
+
+require "placeos-driver/interface/powerable"
+require "placeos-driver/interface/muteable"
+require "placeos-driver/interface/switchable"
+
+class Samsung::Displays::MDCProtocol < PlaceOS::Driver
+  include Interface::Powerable
+  include Interface::Muteable
+
+  INDICATOR = 0xAA_u8
+
+  enum Inputs
+    Vga           = 0x14 # pc in manual
+    Dvi           = 0x18
+    Dvi_Video     = 0x1F
+    Hdmi          = 0x21
+    Hdmi_Pc       = 0x22
+    Hdmi2         = 0x23
+    Hdmi2_Pc      = 0x24
+    Hdmi3         = 0x31
+    Hdmi3_Pc      = 0x32
+    Hdmi4         = 0x33
+    Hdmi4_Pc      = 0x34
+    Display_Port  = 0x25
+    Dtv           = 0x40
+    Media         = 0x60
+    Widi          = 0x61
+    Magic_Info    = 0x20
+    Whiteboard    = 0x64
+  end
+  include PlaceOS::Driver::Interface::InputSelection(Inputs)
+
+  # Discovery Information
+  tcp_port 1515
+  descriptive_name "Samsung MD, DM & QM Series LCD"
+  generic_name :Display
+
+  # Markdown description
+  description <<-DESC
+  For DM displays configure the following 1:
+
+  1. Network Standby = ON
+  2. Set Auto Standby = OFF
+  3. Set Eco Solution, Auto Off = OFF
+
+  Hard Power off displays each night and hard power ON in the morning.
+  DESC
+
+  default_settings({
+    display_id: 0,
+    rs232_control: false
+  })
+
+  @id : Int32 = 0
+  @rs232 : Bool = false
+  @blank : Input?
+  @previous_volume : Int32 = 50
+  # Meta data for inquiring interfaces
+  @input_stable : Bool = true
+  @input_target : Input = Input::Hdmi
```

Comment:
Defaulting to an input here will result in a connected display being forced to HDMI after the first poll, despite this not being requested by the driver. May be worth making this a nilable field.